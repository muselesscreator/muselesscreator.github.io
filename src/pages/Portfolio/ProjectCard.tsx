import { FaEye } from 'react-icons/fa';

import { StrictDict } from '@muselesscreator/strict-dict';

import { ProjectData } from './types';


export const classNames = StrictDict({
  img: 'project-img',
  iconBox: 'project-item-icon-box',
  title: 'project-title',
  category: 'project-category',
}) as Record<string, string>;

const ProjectCard = ({
  title,
  description,
  category,
  image,
  projectUrl,
}: ProjectData) => (
  <a href={projectUrl} target="_blank" rel="noopener noreferrer">
    <figure className={classNames.img}>
      <div className={classNames.iconBox}>
        <p>{description}</p>
        <FaEye />
      </div>
      <img src={image} alt={title} loading="lazy" />
    </figure>
    <h4 className={classNames.title}>{title}</h4>
    <p className={classNames.category}>{category}</p>
  </a>
);

export default ProjectCard;

