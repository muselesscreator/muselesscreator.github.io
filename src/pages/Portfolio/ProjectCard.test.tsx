import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';
import { shallow } from '@muselesscreator/react-shallow-snapshot';
import ProjectCard, { classNames } from './ProjectCard';

vi.mock('react-icons/fa', () => ({ FaEye: 'FaEyeIcon' }));

const project = {
  title: 'Project Title',
  description: 'Project Description',
  category: 'Project Category',
  githubUrl: 'github-url',
  githubRepo: 'github-repo',
  projectUrl: 'project-url',
};

describe('ProjectCard', () => {
  test('snapshot', () => {
    const el = shallow(<ProjectCard  {...project} />);
    expect(el.snapshot).toMatchSnapshot();
    const iconBox = el.instance.findByClassName(classNames.iconBox)[0];
    const title = el.instance.findByClassName(classNames.title)[0];
    const category = el.instance.findByClassName(classNames.category)[0];
    expect(iconBox.findByType('p')[0].children[0].el).toEqual(project.description);
    expect(title.children[0].el).toEqual(project.title);
    expect(category.children[0].el).toEqual(project.category);
  });
});
