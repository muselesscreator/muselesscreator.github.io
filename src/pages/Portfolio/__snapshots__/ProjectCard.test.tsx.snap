// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectCard > snapshot 1`] = `
<a
  href="project-url"
  rel="noopener noreferrer"
  target="_blank"
>
  <figure
    className="project-img"
  >
    <div
      className="project-item-icon-box"
    >
      <p>
        Project Description
      </p>
      <FaEyeIcon />
    </div>
    <img
      alt="Project Title"
      loading="lazy"
    />
  </figure>
  <h4
    className="project-title"
  >
    Project Title
  </h4>
  <p
    className="project-category"
  >
    Project Category
  </p>
</a>
`;
